common/djangoapps/student/views.py

```diff
@@ -728,6 +732,17 @@ def dashboard(request):
 
     enterprise_message = get_dashboard_consent_notification(request, user, course_enrollments)
 
+    enterprise_customer = enterprise_customer_for_request(request)
+    if enterprise_customer:
+        consent_required_courses = frozenset(
+            enrollment.course_id for enrollment in course_enrollments
+            if consent_needed_for_course(request, request.user, str(enrollment.course_id), True)
```

Comment:
I'm concerned about making an API request here for each CourseEnrollment. It would be nice if you send a list of course IDs to the consent API endpoint.

Reply:
Would this require a new API to be exposed? I didn't see one that took a list.
Also, the call to consent_needed_for_course caches the result in the session so it's only hitting the endpoint the first time this is called for a user, or on subsequent calls if consent is required and they haven't consented yet.

Reply:
Yes, this would require modifying the consent endpoint.

Reply:
I'd propose making that change in another PR as it shouldn't be an issue for the normal enterprise customer use case. I filed this ticket to track: https://openedx.atlassian.net/browse/WL-1315

Reply:
@bill-filler So, it turns out this is causing a bunch of Splunk alerts on a daily basis for us. Take a look at this query in Splunk:

`index=prod-edx source="/edx/var/log/lms/edx.log" "HttpClientError: Client Error 429"`

The `enterprise-learner` API endpoint is getting throttled for learners with many course enrollments. I kind of want to revert this change. What do you think we should do?

Also, see https://openedx.atlassian.net/browse/ENT-727.